Wonka-Conference : Add Home and About Pages - Mobile & Desktop Version

OmarRamoun opened the pull request:
## Changes Done :recycle:

A Conference Website for the Wonka Conference for family health. Built from scratch using HTML5, CSS3, JS, and mostly the Bootstrap framework, taking into account the requirements specified below. Link to the video description here.

### **Linters:**
#### - :heavy_plus_sign: Added HTML/CSS and JavaScript linters.
### **HTML:**
#### - :heavy_plus_sign: Added an HTML Home Page (index.html)
#### - :heavy_plus_sign: Added an HTML About Page (about/index.html)
### **CSS:**
#### - :heavy_plus_sign: Added CSS styling to Home Page and About Page (style.css)
#### - :heavy_plus_sign: Added CSS media query to manage and make the entire website responsive for different sizes.
### **JavaScript:**
#### - :heavy_plus_sign: Added JS Scripts to make a dynamic HTML about speakers data when the page load's (speakers-data.js)
#### - :heavy_plus_sign: Added Js Script functionality to my own mobile navbar (menu-options.js)
### **Extras:**
#### - :heavy_plus_sign: Added FontAwesome 5 icons
#### - :heavy_plus_sign: Added custom background images on HomePage and AboutPage
#### - :heavy_plus_sign: Added customized content about E3 Gaming Event.
#### - :heavy_plus_sign: Added Bootstrap to desktop navbar version.
#### - :heavy_plus_sign: Added and Designed own mobile navbar
### **About:**
#### - :heavy_plus_sign: Added Licenses and Credits to Cindy Shin ( Owner of Template)
#### - :heavy_plus_sign: Added Detail and Descriptive README.me file
#### - :heavy_plus_sign: Added Live Server Demo: Try It!
#### - :heavy_plus_sign: Added Informative video about my job: Project Video

### Link to Loom Video
- [link](https://www.loom.com/share/a65e01ea771447edb23a07935a095f23)

Commit:
Add information to the plan file

Commit:
Add information about tools used

- add to tools in manifesto

Commit:
Update README information

- Add Project Information to the readme file

Commit:
:recycle: Update package information

- Update package.json and package-lock.json files

Commit:
:recycle: Update contents of package files

- Update package files after local installation of the packages

Commit:
Merge branch 'feature/update-repo-files' into develop

Commit:
Add assets folder

- Add fonts used

Commit:
Add images folder

Commit:
Merge branch 'feature/add-assets' into develop

Commit:
Add website description

- Add detailed description of the conference

Commit:
Add stylesheets and title

- Add a link to assets folder CDN
- Add a link to images folder CDN
- Add Title to the page

Commit:
Add boostrap cdn links with formating

- Add vertical space to seprate head child elements logically
- Add link to boostrap minified stylesheet
- Add link to boostrap minified javascript
- Add link to jquery minified javascript
- Add link to boostrap tooltip script

Commit:
Add comments to the head

- Add comment for the stylesheets
- Add comment for bootstrap section

Commit:
Merge branch 'feature/add-head-info' into feature/add-page-structure

Commit:
Add structure of the header

Commit:
Add home and menu icons to images folder

Commit:
Add wrapper class to header

Commit:
Remove redundent divs

Commit:
Merge branch 'add-header-section' into feature/add-intro-section

Commit:
Add hero basic html structure

Commit:
Add wrapper class to hero section

Commit:
Replace time element with p

Commit:
Move hero up and use div tags

- Move hero section up to be a child of the intro section - Replace the section tag with a div tag

Commit:
Merge branch 'add-hero-section' into feature/add-intro-section

Commit:
Merge branch 'feature/add-intro-section' into feature/add-page-structure

Commit:
Add program section basic structure

- Add container structure of the program section - Add heading for the section

Commit:
Add cards basic structure

- Add structure for cards container with basic card template strcure

Commit:
Add footer of the cards section

Commit:
Add basic structure of speakers section

Commit:
Add cards grid basic structure

- Add strutre for cards grid with basic card template structure

Commit:
Add footer of speakers section

Commit:
Add partners section basic structure

Commit:
Add list of partners of the conference

Commit:
Add basic structure of the footer section

Commit:
Add copyrights section to footer

Commit:
Merge branch 'feature/add-program-section' into feature/add-page-structure

Commit:
Merge branch 'feature/add-speakers-section' into feature/add-page-strcutre

Commit:
Merge branch 'feature/add-partners-section' into feature/add-page-structure

Commit:
Merge branch 'feature/add-footer-section' into feature/add-page-structure

Commit:
Merge branch 'feature/add-page-structure' into develop

Commit:
Add links to styleshets

- Add link to style.css
- Add link to layout.css

Commit:
Replace div tags with nav tags

- Make the navbar more semantic by adding a nav tag

Commit:
Add side navbar structure, layout, & style

Commit:
Remove hamburger menu border

Commit:
Modify the reset font-size

Commit:
Add margin top class to the layout

Commit:
Merge branch 'feature/style-header-section' into feature/style-intro-section

Commit:
Add checker background to the intro section

Commit:
Update hero section classes

- for better styling and layout

Commit:
Remove redundent code from the footer

Commit:
Update default font-family and font-size

- Edit the reset css file to have different default font properties

Commit:
Add top padding rule class to the layout

Commit:
Update the color variables

- to match design guidlines

Commit:
Add general purpose custom classes

Commit:
Update web font stylesheet

Commit:
Add background image to the images folder

Commit:
Merge branch 'feature/style-intro-section' into feature/add-page-styles

Commit:
Add scroll-opacity effect to navbar

- Increase opacity up to 100% on screen scroll using a js scirpt

Commit:
Add background to program section

- Add a css class to add dark background.
- Add a css class to add pattern background image.

Commit:
Remove redundent margin declaration

from style.css file

Commit:
Update plus svg settings

Commit:
Add padding in the y axis to the program section

Commit:
Update cards section alongside the heading

- Add bootstrap styling classes to different elements of the section
- Add other 4 duplicate cards
- Add bootstrap classes to the heading of the programs section

Commit:
Add styling to the footer section of the programs

Commit:
Add general 30% width class to the stylesheet

Commit:
Update 'bg' classes in the style.css file

- Add accent background class
- Add accent background hover class
- Add semi-transparent background class
- Make .bg-pattern class declaration more specific by adding a background-image

Commit:
Add .heading class to style.css

- Add orange underline to the heading by:
  - Adding a class to position the heading
  - Adding a pseudo-element and style it to be underlined

Commit:
Merge branch 'feature/style-program-section' into feature/add-page-styles

Commit:
Add bootstrap classes to the speakers section

- Add bootstrap classes to speacker's section and it's elements to make set the style and layout of it

Commit:
Merge branch 'feature/style-speakers-section' into feature/add-page-styles

Commit:
Add dark background and margin to the partners section

Commit:
Add bootstrap classes to the children of the partners section

- to make layout and style them

Commit:
Merge branch 'feature/style-partners-section' into feature/add-page-styles

Commit:
Replace the CC SVG

with a bigger one with high resolution

Commit:
Add bootstrap styles to footer section

Commit:
Merge branch 'feature/style-footer-section' into feature/add-page-styles

Commit:
Merge branch 'feature/add-page-styles' into develop

Commit:
Convert the Pages folder into about folder

to use for storing the html file representing the about page

Commit:
Add all head section detials and tags

Commit:
Update paths to stylesheets

Commit:
Merge branch 'feature/add-head-info' into feature/add-page-structure

Commit:
Add header section with navbar

Commit:
Change backgroun of intro section

Commit:
Add hero section with bootstrap classes

Commit:
Add styling classes to style.css

Commit:
Delete pages folder for its redundency

Commit:
Merge branch 'feature/add-intro-section' into feature/add-page-structure

Commit:
Add about section with bootstrap classes

Commit:
Merge branch 'feature/add-about-section' into feature/add-page-structure

Commit:
Add show section with bootstrap classes

Commit:
Add css classes to styles file

Commit:
Merge branch 'feature/add-show-section' into feature/add-page-structure

Commit:
Add parners section with bootstrap classes

Commit:
Add footer section to about page

Commit:
Fix a broken image link of the footer

Commit:
Merge branch 'feature/add-page-structure' into feature/add-about-page

Commit:
Merge branch 'feature/add-about-page' into develop

Commit:
Update link to about page in the navbar

Commit:
Add desktop css file

Commit:
Add link to desktop layout in home and about

Commit:
Update the navbar bootstrap classes for Desktop version

Commit:
Change the color of social media icons to white

Commit:
Add desktop classes to the desktop stylesheet

Commit:
Add class to remove sub navbar on mobile version

Commit:
Add bootstrap classes to grid columns

- Add bootstrap class to container breakpoint column on large screens
- Add bootstrap class to heading breakpoint column on medium screens
- Add bootstrap class to paragraph breakpoint column on medium screens
- Add bootstrap class to lower third breakpoint column on medium screens

Commit:
Add large screen class to desktop view

- Adds margin left and top class

Commit:
Update breakpoints on different sizes for navbar and hero

Commit:
Add margin right to th margin class

Commit:
Change font size of main heading

Commit:
Update script with correct class name

Commit:
Merge branch 'feature/desktop-intro' into feature/desktop-version

Commit:
Change Flex direction on large

- Change layout direction of the cards container to row.
- Change layout direction of the card to a column

Commit:
Merge branch 'feature/desktop-program' into feature/desktop-version

Commit:
Add a card dummy speaker card

Commit:
Display cards in to different columns

Commit:
Merge branch 'feature/desktop-speakers' into feature/desktop-version

Commit:
Change width of the partners images

Commit:
Add class with max-width

Commit:
Add max width class to small screens

Commit:
Add classes to images to change them size

Commit:
Add a class to change the max-width

Commit:
Add a class to the layout stylesheet

Commit:
Merge branch 'feature/desktop-partners' into feature/desktop-version

Commit:
Add footer class to the footer for selecting img

- Add footer class to the footer element
- Use footer class as a selector to select img child and change their size on desktop screens

Commit:
Add bootstrap class to remove bottom margin of a paragraph

Commit:
Merge branch 'feature/desktop-footer' into feature/desktop-version

Commit:
Merge branch 'feature/desktop-version' into develop

Commit:
Update header section for desktop

Commit:
Update intro section for large screens

Commit:
Update about section for large screens

Commit:
Update show section for large screens

Commit:
Update partners section for large screens

Commit:
Update footer section for large screens

Commit:
Add link to scroll script

Commit:
Merge branch 'desktop-about' into develop

Commit:
Add script for adding speakers dynamically

Commit:
Update speakers columns to use flex to add gaps

Commit:
Add link to speakers script

- Add link to speakers.js which a script that adds speakers dynamically to the list of speakers

Commit:
Merge branch 'feature/add-dynamics' into develop

Commit:
Remove redundent comments

Commit:
Change column size for hero heading on large screens

Commit:
Adjust large screen size from 700px to 768px

Commit:
Change column size for small screens

- change column size to 12 instead of 2 and add a margin-bottom to the left column

Commit:
Add a new class for the 13px size

Commit:
Add a none class to the footer and remove href

Commit:
Add show/hide functionality to show speakers btn

- Add a variable to watch for window width
- Add a selector for the button
- Extract the display on load for loop into a function
- Remove the children of the `.speakers` section on load or resize
- Add resize function to change the number of speaker cards on call
- Add toggle function to change the number of speaker cards on button click
- Add load and resize event listeners for the window
- Add click event listener for the button

Commit:
Change breakpoint of left column of speakers section

Commit:
Change breakpoint of footer  of speakers section

Commit:
Change breakpoint of footer of partners section

Commit:
change copyright image width

Commit:
Add a new width class for 20%

Commit:
Fix broken links for about page

Commit:
Add onClick Event to logo

Commit:
Merge branch 'bugfix/fix-all-website-bugs' into develop

Commit:
Fix all webhint linter errors

Commit:
Fix stylelint linter errors

Commit:
Fix eslint linter errors

Commit:
Fix all Webhint linter errors

Commit:
Fix eslint and stylelint errors

Commit:
Fix linter errors

Commit:
Delete Redundent demo files

emmanuellekamwa commented on README.md:
```diff
@@ -2,28 +2,27 @@
+> Wonka Converence 2021 Made with Vanilla HTML5, CSS3, & JS. The Website Contains A Homepage With Information About The Main Program Of The Conference Speakers And Partners And An About Page With Information About The Conference.
 
 ![screenshot](images/app_screenshot.png)
 
```
- [x] Hello. Please this screenshot is not necessary here. You can choose to either replace it with an actual screenshot of your project or you can just delete it.
- [x] I have noticed that you have forgotten to add the link to your video presentation. Please add it.

emmanuellekamwa commented on README.md:
```diff
@@ -87,27 +84,20 @@ To get a local copy up and running follow these simple example steps.
-- Hat tip to anyone whose code was used
 - Inspiration
-- etc
+- My Passion for Programming
```
- [x] In this section, don't forget to add credit to [Cindy Shin](https://www.behance.net/adagio07) the [author of the original design](https://www.behance.net/gallery/29845175/CC-Global-Summit-2015).

emmanuellekamwa commented on about/index.html:
```diff
@@ -0,0 +1,146 @@
+        2021 WONCA World Conference. All rights reserved.
+      </p>
+    </div>
+  </footer>
```
- [x] The footer contains a logo aligned to the left and text aligned to the right but on your page, the logo and the text are aligned on the left. Please fix that on both pages.

emmanuellekamwa commented on index.html:
```diff
@@ -6,15 +6,181 @@
+        <a class="d-none text-center" href="#">Learn More Information</a>
+      </div>
+    </div>
+  </section>
```
- [x] In mobile view, the icons of this section are too big. Please try and reduce their sizes.

emmanuellekamwa commented on index.html:
```diff
@@ -6,15 +6,181 @@
+        <a class="btn btn-custom text-capitalize text-light mx-auto p-3 bg-accent">more speakers</a>
+      </div>
+    </div>
+  </section>
```
- [x] On this section, I suggest you should remove the borders around the divs and add some background behind the images.

emmanuellekamwa reviewed (changes requested):
**STATUS: CHANGES REQUESTED 🛠🛑**
Hello @MrRamoun 👋🏽
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._👇🏽👀

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Commit:
Update screenshot of the project

Commit:
Add credits to original designer

Commit:
Align footer text to right and icon to left

Commit:
Decrease the size of the Main Program section

- Add a new class to all img elements to control their width
- Make width fixed at 30px for desktop view
- Make width adjustable according to window width for mobile view

Commit:
Adjust alignment for footer children for about page

Commit:
Add background image to the speakers and remove borders

Commit:
Change alt text of speaker images

Commit:
Merge branch 'bugfix/apply-review-tips' into develop

george-shammar reviewed (approved):
Hi @MrRamoun , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

:heavy_check_mark: This is truly impressive and thank you for making the required changes. :+1: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
